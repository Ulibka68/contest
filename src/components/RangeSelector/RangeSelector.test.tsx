import React from "react";
import { shallow } from "enzyme";
import {
  RangeSelector,
  RangeWindow,
  RangeMask,
  RangeHanlder,
} from "./RangeSelector";
import * as faker from "faker";
import { matchers } from "jest-emotion";
import toJson from "enzyme-to-json";

const setPosition = jest.fn();
expect.extend(matchers);

describe("RangeSelector behaviour check", () => {
  it("is RangeSelector with stable date to match snapshot", () => {
    const wrapper = shallow(
      <RangeSelector
        position={{ left: 10, right: 10 }}
        setPosition={setPosition}
      />
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it("RangeMask with isNight have night theme color", () => {
    const wrapper = shallow(<RangeMask isNight={true} />);
    expect(toJson(wrapper.render())).toHaveStyleRule(
      "background",
      "rgba(48,66,89,0.6)"
    );
  });
  it("RangeMask with isLeft have border-top-left-radius", () => {
    const wrapper = shallow(<RangeMask isLeft={true} />);
    expect(toJson(wrapper.render())).toHaveStyleRule(
      "border-top-left-radius",
      "8px"
    );
  });
  it("RangeMask without isLeft have border-top-right-radius", () => {
    const wrapper = shallow(<RangeMask />);
    expect(toJson(wrapper.render())).toHaveStyleRule(
      "border-top-right-radius",
      "8px"
    );
  });

  it("RangeHanlder with isNight have night theme color", () => {
    const wrapper = shallow(<RangeHanlder isNight={true} />);
    expect(toJson(wrapper.render())).toHaveStyleRule("background", "#56626D");
  });

  it("RangeHanlder with isLeft have border-top-left-radius", () => {
    const wrapper = shallow(<RangeHanlder isLeft={true} />);
    expect(toJson(wrapper.render())).toHaveStyleRule(
      "border-top-left-radius",
      "8px"
    );
  });

  it("RangeHanlder without isLeft have border-top-right-radius", () => {
    const wrapper = shallow(<RangeHanlder />);
    expect(toJson(wrapper.render())).toHaveStyleRule(
      "border-top-right-radius",
      "8px"
    );
  });
});
