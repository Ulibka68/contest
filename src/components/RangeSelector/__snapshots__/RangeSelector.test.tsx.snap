// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSelector behaviour check is RangeSelector with stable date to match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-r9hute",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "css-lg4iqz",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "css-s21fp2",
            "style": "width:10px",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "css-pyierp",
              "style": "width:10px",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "css-nerg3i",
                "style": "left:10px;right:10px",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "css-1p4fbz0",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "css-krxh19",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "css-krxh19",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "css-1p4fbz0",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "style": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "css-s21fp2",
          "style": "width:10px",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "css-pyierp",
            "style": "width:10px",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "css-nerg3i",
              "style": "left:10px;right:10px",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "css-1p4fbz0",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "css-krxh19",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "css-krxh19",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "css-1p4fbz0",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "css-lg4iqz",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "css-pyierp",
          "style": "width:10px",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "css-nerg3i",
            "style": "left:10px;right:10px",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-1p4fbz0",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "css-krxh19",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "css-krxh19",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "css-1p4fbz0",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "css-s21fp2",
            "style": "width:10px",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "css-lg4iqz",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "css-nerg3i",
          "style": "left:10px;right:10px",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-1p4fbz0",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "css-krxh19",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "css-krxh19",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "css-1p4fbz0",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "css-pyierp",
            "style": "width:10px",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "css-s21fp2",
              "style": "width:10px",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "css-lg4iqz",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
